### ISSUE TITLE
Cross compile Tensorflow with host protoc

### ISSUE THREAD

**Issue by jasaw** [tensorflow/tensorflow#56859]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Build/Install

### Source

source

### Tensorflow Version

2.9.1

### Custom Code

No

### OS Platform and Distribution

Linux

### Current Behaviour?

```shell
I'm cross compiling Tensorflow for aarch64 target on my x86-64 host.
Bazel pulls in protobuf and uses the aarch64 compiler to compile protoc and then tries to execute it on my x86-64 host which fails.
```

### Standalone code to reproduce the issue

```shell
How do I tell it to use my host protoc?
This is my current build command.
`bazel build --crosstool_top=//third_party/toolchains/cpus/aarch64:toolchain --cpu=arm //tensorflow/tools/pip_package:build_pip_package`
```

### Relevant log output

_No response_</details>

**Comment by mohantym** [comment]
Hi @jasaw !
Can you check the cross-compilation [document ](https://www.tensorflow.org/lite/guide/build_cmake_arm#build_for_aarch64_arm64)from Cmake and let us know whether it works.

**Comment by jasaw** [comment]
@mohantym I'm sorry, I don't quite understand what you want me to try. I'm building Tensorflow using Bazel, which downloads and builds protobuf as part of the process. Do you want me to try building protobuf separately using cmake?

**Comment by mohantym** [comment]
@jasaw 
I understand you are trying to build in arm system But  Crossing compiling with different tool through Bazel is is  prone to lot of efforts and errors . I was suggesting to use CMake instead ( But i don't suggest to build protobuf separately). 

Else you should test the instruction[ from ARM](https://www.tensorflow.org/lite/guide/build_arm) on bazel. 

Thank you!

**Comment by jasaw** [comment]
@mohantym Cmake is only available for Tensorflow lite. I'm trying to cross compile Tensorflow with CUDA support, which means Bazel is the only choice.

**Comment by jasaw** [comment]
@mohantym After adding k8 compiler to my `cc_toolchain_suite`, I managed to get the build to progress further, but now I encounter this error.
>tensorflow-v2.9.1/tensorflow/core/ops/BUILD:46:15: Compiling tensorflow/core/ops/parsing_ops.cc [for host] failed: undeclared inclusion(s) in rule '//tensorflow/core/ops:parsing_ops_op_lib':
this rule is missing dependency declarations for the following files included by 'tensorflow/core/ops/parsing_ops.cc':
  'bazel-out/host/bin/tensorflow/core/example/example.pb.h'
  'bazel-out/host/bin/tensorflow/core/example/feature.pb.h'

Any ideas on what might be wrong?

**Comment by jasaw** [comment]
@mohantym I resolved my previous issue and hit another error. The bazel build rules should already point it to the correct cuda library with `cudaGetErrorString` function? What's wrong this time?

>ERROR: tensorflow-v2.9.1/tensorflow/core/kernels/BUILD:4365:18: output 'tensorflow/core/kernels/_objs/histogram_op_gpu/histogram_op_gpu.cu.pic.o' was not created
ERROR: tensorflow-v2.9.1/tensorflow/core/kernels/BUILD:4365:18: Compiling tensorflow/core/kernels/histogram_op_gpu.cu.cc failed: not all outputs were created or valid
./tensorflow/core/platform/file_system.h(579): warning: overloaded virtual function "tensorflow::FileSystem::FilesExist" is only partially overridden in class "tensorflow::WrappedFileSystem"
./tensorflow/core/platform/file_system.h(579): warning: overloaded virtual function "tensorflow::FileSystem::CreateDir" is only partially overridden in class "tensorflow::WrappedFileSystem"
./tensorflow/core/platform/env.h(498): warning: overloaded virtual function "tensorflow::Env::RegisterFileSystem" is only partially overridden in class "tensorflow::EnvWrapper"
ptxas fatal   : Unresolved extern function 'cudaGetErrorString'
Target //tensorflow/tools/pip_package:build_pip_package failed to build

**Comment by gadagashwini** [comment]
HI @jasaw, Looks like GCC version problem. Which Gcc and Protobuf version you are using? Thank you!

**Comment by jasaw** [comment]
@gadagashwini I'm using GCC version 7.5.0.
Protobuf is the version that is defined in tensorflow workspace, which is version 3.9.2.

**Comment by jasaw** [comment]
I resolved the issue by using GCC 7.5.0 on my build machine to match my target GCC version and removing nvcc device debug stops ptxas from crashing.
